BSP/Simulator/Src/BSP_ADC.c

```diff
@@ -0,0 +1,129 @@
+/*
+C file that reads the raw ADC data from the CSV file, 
+converts it back to a distance pressed (percentage for now), and stores both.
+We should be able to request both the raw ADC data,
+the distance (percentage for now) and millivolts from this library.
+ */
+
+//NOTE: ADC precision is 12 bits (4096), range 3.3V
+//I'm assuming 3.3V corresponds to a 100% pressed pedal
+
+#include "BSP_ADC.h"
+
+#define FILE_NAME DATA_PATH(PEDALS_CSV)
+/**
+ * @brief   Confirms that the CSV file
+ *          has been created and prints and 
+ *          error message if not
+ * @param   None
+ * @return  None
+ */ 
+void BSP_ADC_Init(void) {
+    // Checking if file exists
+   if(access(FILE_NAME, F_OK) != 0){
+        // File doesn't exist if true
+        perror(PEDALS_CSV);
+        exit(EXIT_FAILURE);
+    }
+    // There really isn't much we have to do in the simulator
+}
+
+/**
+ * @brief   Helper function which returns a 
+ *          milivoltage based on the raw ADC value
+ *          compared to the 3.3 volt range of the 
+ *          pedals' potentiometers
+ * @param   raw ADC value
+ * @return  milivolts 
+ */ 
+int convert_ADC_to_Voltage(int ADCvalue){
+    return ((ADCvalue / 4096) * 3300);
+}
+
+/**
+ * @brief   Helper function that gets
+ *          the number of ADC values 
+ *          present in the csv file,
+ *          each value has to be separated by a comma
+ *          or new line
+ * @param   none
+ * @return  numberOfADCValues, the number of ADC channels connected to the ADC providing raw ADC data
+ */ 
+int number_Values(){
+    int numberOfADCValues = 0;
+    //opening file in read mode
+    FILE *filePointer = fopen(FILE_NAME, "r");
+    //counting how many ADC Values the file has
+    for(char c = getc(filePointer); c!= EOF; c = getc(filePointer)){
+        if(c == '\n' || c == ','){
+            numberOfADCValues++;
+        }
+    }
+    //close the file 
+    fclose(filePointer);
+    return numberOfADCValues;
+}
```

Comment:
I don't think this function is necessary since we can read from a file just until the file ends. So we don't need to know in advance how many entries there are